Source/Dafny/AST/DafnyAst.cs

```diff
@@ -9158,27 +9158,49 @@ public abstract class Expression {
 
     private RangeToken rangeToken;
 
+    protected IToken[] FormatTokens = null;
```

Comment:
I'd add a comment here about what this is for.

Reply:
>I'd add a comment here about what this is for.

Added
```
    // Contains tokens that did not make it in the AST but are part of the expression,
    // Enables ranges to be correct.
```